### PR TITLE
Introduce advise_using_config to enable supplying config file

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/thamos/issues/622

## This introduces a breaking change

- [x] No

### CODE REVIEW

**Comment by fridex** [comment]
Usage:


```python
from thamos.lib import advise_using_config

with open("Pipfile", "r") as f:
    pipfile = f.read()

with open("Pipfile.lock", "r") as f:
    pipfile_lock = f.read()

advise_using_config(
    pipfile=pipfile,
    pipfile_lock=pipfile_lock,
    config=".thoth.yaml",  # can be also directly a YAML/JSON with the configuration file content
    runtime_environment_name="rhel:8",
)
```

File: thamos/config.py [hunk 1]
```diff
@@ -112,19 +112,31 @@ def config_file_exists(self) -> bool:
...
+    def load_config_from_string(self, config_str: str) -> None:
+        """Load configuration from a string."""
+        if int(os.getenv("THAMOS_CONFIG_EXPAND_ENV", 0)):
+            _LOGGER.info("Expanding configuration file based on environment variables")
+            config_str = config_str.format(**os.environ)
+
+        self._configuration = yaml.safe_load(config_str)
+
+    def load_config_from_file(self, config_path: str) -> None:
+        """Load configuration from a string."""
```
Comment (pacospace) [inline thamos/config.py:124]: """Load configuration from file using path."""

**Review by pacospace — COMMENTED** [review]
One small comment, otherwise great :100: Thanks!

**Review by pacospace — APPROVED** [review]
:beers: Thanks!

**Pull request merged**